### PR TITLE
microbit: add Makefile entries for target=microbit

### PR DESCRIPTION
**Pull request by deadprogram**
Add Makefile entries for target=microbit

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,6 +47,9 @@ run-blinky2: build/blinky2
 ifeq ($(TARGET),pca10040)
 flash-%: build/%.hex
 	nrfjprog -f nrf52 --sectorerase --program $< --reset
+else ifeq ($(TARGET),microbit)
+flash-%: build/%.hex
+	nrfjprog -f nrf51 --sectorerase --program $< --reset
```
Comment (aykevl) [inline Makefile]: Have you tried using openocd with `-f interface/cmsis-dap.cfg`?
In my experience, nrfjprog only works when the chip has Segger firmware which is non-standard.

Comment (aykevl) [reply Makefile]: It's in microbit.json:

    openocd -f interface/cmsis-dap.cfg -f target/nrf51.cfg -c 'program {hex} reset exit'

Translated to Makefile syntax (see bluepill/stm32):

    openocd -f interface/cmsis-dap.cfg -f target/nrf51.cfg -c 'program $< reset exit'

Comment (deadprogram) [reply Makefile]: That works great, and no firmware update required is better. Still seems like adding to the makefile for easy loading would be a good thing. Should I update this PR accordingly?

**Comment by deadprogram** [comment]
The flashing part is still very useful, as far as having wrappers around the params for different boards.

Perhaps a solution could be a small separate command line program ala `tinygo-flasher`? Another might be some refactors in the big makefile in the main project to make the flashing part more modular to serve as a starting point for people to write their own project-specific makefile just by copying/pasting those parts of the makefile `flash_microbit` tasks or whatever.

File: Makefile (hunk 1 above)
Comment (aykevl) [reply Makefile]: Yes, please. Then I'm sure it works for both of the firmwares :)

**Comment by aykevl** [comment]
What's exactly the use of the Makefile for you?
For me, it's mostly the length of the command:

    make flash-blinky1 TARGET=arduino
    tinygo flash -target arduino -size short examples/blinky1

I have ideas how to fix that, but am not sure what the correct way is. One way could be to specify the default target in the package itself or in a global configuration file. Another simple one is to use the current directory as the package name if none is specified on the command line, like the `go` command does. So that this would work from within a given project:

    tinygo flash

**Comment by deadprogram** [comment]
I like the idea of `tinygo flash`. :+1:

**Comment by aykevl** [comment]
Merged in 269bef5494cb928eeadb5fdc99d417f6e6bfcb03.

**Pull request closed without merging**